server/service/src/sync/central_data_synchroniser_v6.rs

```diff
@@ -157,14 +168,25 @@ impl SynchroniserV6 {
             .map(SyncRecordV6::from)
             .collect();
 
+            if records.is_empty() {
+                // No v6 records to push, update cursor and continue to next batch
+                if let Some(last_pushed_cursor_id) = last_pushed_cursor {
+                    cursor_controller.update(connection, last_pushed_cursor_id as u64 + 1)?
+                };
```

Comment:
Should I put this logic in a callback? Also used below 😱

Reply:
Ooh, I am afraid of indentation, but maybe in this case it would be better to do below if !records.is_empty()